docs/sources/flow/concepts/clustering.md

```diff
@@ -0,0 +1,243 @@
+or an existing node going away, all participating components locally
+recalculate target ownership and rebalance the number of targets they’re
+scraping without explicitly communicating ownership over the network.
+
+The agent makes use of a fully-local consistent hashing algorithm to distribute
+targets, meaning that on average only ~1/N of the targets are redistributed.
+This is in contrast to hashmod sharding where up to 100% of the targets could
+be reassigned to another node and possibly cause system instability.
+
+As such, target auto-distribution not only allows to dynamically scale the
+number of agents to distribute workload during peaks, but also provides
+resiliency, since in the event of a node going away, its targets are
+automatically picked up by one of their peers. Again, this is in contrast to
+hashmod sharding which requires running multiple replicas of each shard for HA.
+
+The components who can make use of target auto-distribution are the following:
+- [prometheus.scrape][]
+- [pyroscope.scrape][]
+- [prometheus.operator.podmonitors][]
+- [prometheus.operator.servicemonitors][]
+
+These components can opt-in to auto-distributing targets between nodes by
+defining the `clustering` block like this:
+```river
+prometheus.scrape "default" {
+    clustering {
+      enabled = true
+    }
+    ...
+}
+```
+
+## Local clustering example
+
+The following commands bootstrap a three-node local cluster
+```
+# node-a (seed node)
+$ ./build/grafana-agent-flow \
+    --server.http.listen-addr=0.0.0.0:12345 \
+    --cluster.enabled --cluster.node-name=node-a\
+    --cluster.advertise-address=localhost:12345 \
+    run local-clustering.river
+
+# node-b
+$ ./build/grafana-agent-flow \
+    --server.http.listen-addr=0.0.0.0:12346 \
+    --cluster.enabled --cluster.node-name=node-b \
+    --cluster.advertise-address=localhost:12346 \
+    --cluster.join-addresses=localhost:12345 \
+    run local-clustering.river
+
+# node-c
+$ ./build/grafana-agent-flow \
+    --server.http.listen-addr=0.0.0.0:12347 \
+    --cluster.enabled --cluster.node-name=node-c \
+    --cluster.advertise-address=localhost:12347 \
+    --cluster.join-addresses=localhost:12345,localhost:12346 \
+    run local-clustering.river
+```
+
+The clustering UI page of each node will provide information about the status
+of its peers and the details page of the each node's `prometheus.scrape`
+component will point to which target is being scraped by that node in its
+Debug Info.
+
+## Helm chart clustering example
+
+The easiest way to deploy an agent cluster is by making use of our
+[Helm chart][]. Here's an example of how to achieve that.
+
+The following `values.yaml` file deploys anStatefulSet for metrics
+collection. It makes use of a [headless service][] to retrieve the IPs of the
+agent pods for the `--cluster.join-addresses` argument, as well as an
+[Horizontal Pod Autoscaler][] (HPA) for dynamically matching demand.
+
+```yaml
+--- clustering-values.yaml ---
```

Comment:
Users might accidentally paste this in, I think we should take the filename header out of both code blocks.